src/mpi/coll/igather/igather_inter_sched_short.c

```diff
@@ -26,7 +26,10 @@ int MPIR_Igather_inter_sched_short(const void *sendbuf, int sendcount, MPI_Datat
     remote_size = comm_ptr->remote_size;
     local_size = comm_ptr->local_size;
 
-    if (root == MPI_ROOT) {
+    if (root == MPI_PROC_NULL) {
```

Comment:
All processes need to specify the same root value. Supplying `MPI_PROC_NULL` surely is not allowed by the MPI standard, is it?

Reply:
Oh this is inter-communicator, now I see.

Reply:
Yeah, just for the record/information, here is the quote from standard:
```
For this, the root process uses the special root value MPI_ROOT; all other processes in the same group as the root use MPI_PROC_NULL.
```